### ISSUE TITLE
"cache_if" feature

### ISSUE THREAD

**Issue by jeromedalbert** [stackshareio/graphql-cache#56]
Similar to Rails views:

```ruby
- cache_if (current_user.nil?) do
```

And [actionpack-action_caching](https://github.com/rails/actionpack-action_caching):

```ruby
caches_action :show, if: -> { current_user.nil? }
```

It would be nice to be able to do something like this in graphql-cache:

```ruby
field :calculated_field, Int, cache: { if: -> { current_user.nil? } }
```

Although the double-brackets might feel awkward, so it might look cleaner to do 

```ruby
field :calculated_field, Int do
  cache if: -> { current_user.nil? }
end
```

(which is related to my comment on #52 where a suggestion was to be able to have `cache` on its own line, which I believe is not currently possible)

**Comment by drewlustro** [comment]
I agree this would be nice. On our site we'd like to disable the cache for logged in users.
